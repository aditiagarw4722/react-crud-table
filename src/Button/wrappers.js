import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-button');

export const Btn = block('button');

Btn.displayName = 'Button';

export default { Btn };
