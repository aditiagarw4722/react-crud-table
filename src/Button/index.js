import React from 'react';
import { Btn } from './wrappers';

const Button = (props) => <Btn {...props} />;

export default Button;
