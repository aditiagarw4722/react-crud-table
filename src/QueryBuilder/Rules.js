import React from 'react';
import PropTypes from 'prop-types';
import { defaultRuleRender } from './helpers';
import Button from '../Button';
import { NO_OP } from '../helpers';

const Rules = ({ queryRules, onRuleRemoved, renderRule }) => (
  <div>
    {queryRules.map((rule) => (
      <div key={`${rule.field}:${rule.condition}`}>
        {renderRule(rule)}
        {'  '}
        <Button
          onClick={() => {
            onRuleRemoved(rule);
          }}
          modifiers="negative,circular"
        >
          X
        </Button>
      </div>
    ))}
    <br />
  </div>
);

Rules.propTypes = {
  queryRules: PropTypes.instanceOf(Array),
  onRuleRemoved: PropTypes.func,
  renderRule: PropTypes.func,
};

Rules.defaultProps = {
  renderRule: defaultRuleRender,
  queryRules: [],
  onRuleRemoved: NO_OP,
};

export default Rules;
