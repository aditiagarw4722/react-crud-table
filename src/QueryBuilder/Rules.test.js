import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';

import Button from '../Button';
import Rules from './Rules';

const renderer = new ShallowRenderer();

describe('Rules', () => {
  const props = {
    queryRules: [
      {
        field: 'name',
        condition: 'EQUALS_TO',
      },
    ],
    renderRule: jest.fn((x) => `${x.field} ... ${x.condition}`),
    onRuleRemoved: jest.fn(),
  };

  it('renders as expected when no props are passed', () => {
    const result = renderer.render(<Rules />);
    expect(result).toMatchSnapshot();
  });

  it('renders as expected with the given props', () => {
    const result = renderer.render(<Rules {...props} />);
    expect(result).toMatchSnapshot();
  });

  it('notifies when a rule is removed', () => {
    const result = create(<Rules {...props} />);

    result.root.findByType(Button).props.onClick();

    expect(props.onRuleRemoved).toHaveBeenCalledTimes(1);
    expect(props.onRuleRemoved).toHaveBeenCalledWith(props.queryRules[0]);
  });
});
