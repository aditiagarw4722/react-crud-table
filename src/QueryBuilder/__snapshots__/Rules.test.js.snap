// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules renders as expected when no props are passed 1`] = `
<div>
  <br />
</div>
`;

exports[`Rules renders as expected with the given props 1`] = `
<div>
  <div>
    name ... EQUALS_TO
      
    <Button
      modifiers="negative,circular"
      onClick={[Function]}
    >
      X
    </Button>
  </div>
  <br />
</div>
`;
